lib/puppet/daemon.rb

```diff
@@ -88,8 +86,7 @@ def reexec
   end
 
   def reload
-    return unless agent
-    agent.run({:splay => false})
+    @agent.run({:splay => false})
```

Comment:
since we have an `attr_reader` for agent we can keep the same `agent.run({:splay => false}`